src/main/groovy/com/twcable/gradle/cqpackage/CreatePackageTask.groovy

```diff
@@ -0,0 +1,254 @@
+    protected void copy() {
+        this.from(contentSrc)
+
+        // delaying until now to set up these sections because they need the "final" versions
+        // of some of the properties of this task
+        addVaultDefinition(contentSrc)
+        addBundles(bundleInstallRoot)
+        addExclusions(fileExclusions)
+
+        super.copy()
+    }
+
+
+    static enum CopyBundlesMode {
+        ALL, PROJECT_ONLY, NON_PROJECT_ONLY, NONE
+    }
+
+
+    Configuration getConfiguration() {
+        if (configuration != null) return configuration
+
+        return CqPackagePlugin.cqPackageDependencies(project)
+    }
+
+
+    private addExclusions(Collection<String> fileExclusions) {
+        fileExclusions.each {
+            exclude it
+        }
+    }
+
+
+    private addVaultDefinition(@Nonnull File contentSrc) {
+        exclude 'META-INF/vault/definition/.content.xml'
+        exclude 'META-INF/vault/properties.xml'
+
+        into('META-INF/vault') {
+            from new File(contentSrc, 'META-INF/vault/properties.xml')
+            expand(project.properties)
+        }
+
+        into('META-INF/vault/definition') {
+            from new File(contentSrc, 'META-INF/vault/definition/.content.xml')
+            expand(project.properties)
+        }
+    }
+
+
+    private addVaultFilter() {
+        this.mustRunAfter 'addBundlesToFilterXml'
+
+        exclude 'META-INF/vault/filter.xml'
+        into('META-INF/vault') {
+            from project.file("${project.buildDir}/tmp/filter.xml")
+        }
+    }
+
+
+    private void addBundles(@Nonnull String bundleInstallRoot) {
+        if (copyBundlesMode == NONE) return // nothing to do
+
+        this.into("jcr_root${bundleInstallRoot}") { CopySpec spec ->
+            final Collection<File> files
+            switch (copyBundlesMode) {
+                case ALL: files = allBundleFiles(project, configuration); break
+                case PROJECT_ONLY: files = projectBundleJarFile(project, configuration); break
+                case NON_PROJECT_ONLY: files = nonProjectDependencyBundleFiles(configuration); break
+                case NONE: return // should be impossible...
+                default: throw new IllegalStateException("Unknown CopyBundlesMode: ${copyBundlesMode}")
+            }
+
+            logger.info "Adding bundles: ${files}"
+            spec.from files
+        }
+    }
+
+    /**
+     * Root location that jar bundles should be installed to.
+     * Should include a prefixing / but not a trailing one.
+     */
+    String getBundleInstallRoot() {
+        return _bundleInstallRoot
+    }
+
+    /**
+     * Root location that jar bundles should be installed to.
+     */
+    void setBundleInstallRoot(String path) {
+        if (path == null || path.isEmpty())
+            throw new IllegalArgumentException("path == null or empty")
+
+        if (path[0] != '/')
+            path = "/${path}"
+
+        _bundleInstallRoot = (path.size() > 1 && path[-1] == '/') ?
```

Comment:
If `path == '/'` then `path.size() == 1`

Adding the `.trim()` to be paranoid.